python_modules/dagster/dagster/_core/definitions/assets.py

```diff
@@ -1683,39 +1687,35 @@ def _asset_specs_from_attr_key_params(
                     # Value here is irrelevant, because it will be replaced by value from
                     # NodeDefinition
                     skippable=False,
+                    # Value here is irrelevant, because it will be replaced by value from partitions_def
```

Comment:
Btw code like this is why I think it is problematic to use `AssetSpec` as a source of truth object.

Reply:
This is just the legacy codepath for when people pass in a bunch of `metadata_by_key` etc. dicts instead of `AssetSpec`s.